platform/kobo/suspend.sh

```diff
@@ -1,16 +1,36 @@
 #!/bin/sh
 export PATH="${PATH}:/sbin:/usr/sbin"
 
+echo "[$(date +'%x @ %X')] Kobo Suspend: BEGIN!"
 # Disable wifi
 if lsmod | grep -q sdio_wifi_pwr ; then
 	wlarm_le -i eth0 down
 	ifconfig eth0 down
 	rmmod -r dhd
 	rmmod -r sdio_wifi_pwr
+	echo "[$(date +'%x @ %X')] Kobo Suspend: Killed WiFi"
 fi
 
 # Go to sleep
-sync
+current_wakeup_count="$(cat /sys/power/wakeup_count)"
+echo "[$(date +'%x @ %X')] Kobo Suspend: Current WakeUp count: ${current_wakeup_count}"
 echo 1 > /sys/power/state-extended
-sleep 2	# Because reasons?
+# NOTE: Sets gSleep_Mode_Suspend to 1. Used as a flag throughout the kernel to suspend/resume various subsystems
+#       cf. kernel/power/main.c @ L#207
+echo "[$(date +'%x @ %X')] Kobo Suspend: Asked the kernel to put subsystems to sleep"
+sleep 2
+echo "[$(date +'%x @ %X')] Kobo Suspend: Waited for 2s because of reasons..."
+sync
+echo "[$(date +'%x @ %X')] Kobo Suspend: Synced FS"
+echo ${current_wakeup_count} > /sys/power/wakeup_count
```

Comment:
Ha, that's cool. It's definitely not an overkill then. I was asking because I didn't see it in the strace log. According to the kernel doc, this is totally the right thing to do. I wonder why nickel is not doing it ;)